src/test/java/org/joshsim/command/RunRemoteCommandTest.java

```diff
@@ -0,0 +1,352 @@
+/**
+ * Unit tests for RunRemoteCommand class.
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.joshsim.command;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+
+/**
+ * Test cases for RunRemoteCommand functionality.
+ */
+public class RunRemoteCommandTest {
```

Comment:
This is great but let's be very careful that this is not actually generating network traffic.